File: technology-lab/java-showcase/src/main/java/no/ab/showcase/WorkEvent.java
package no.ab.showcase;

/** Immutable work event. Validation happens in the compact canonical constructor. */
public record WorkEvent(String id, Type type, int bytes) {

    public enum Type { CREATED, UPDATED, DELETED }

    public WorkEvent {
        if (id == null || id.isBlank()) {
            throw new IllegalArgumentException("id must not be blank");
        }
        if (type == null) {
            throw new IllegalArgumentException("type must not be null");
        }
        if (bytes < 0) {
            throw new IllegalArgumentException("bytes must not be negative");
        }
    }
}

File: technology-lab/kotlin-showcase/src/main/kotlin/no/ab/showcase/WorkEvent.kt
package no.ab.showcase

enum class EventType { CREATED, UPDATED, DELETED }

/** Immutable work event; invalid values are rejected at construction time. */
data class WorkEvent(val id: String, val type: EventType, val bytes: Int) {
    init {
        require(id.isNotBlank()) { "id must not be blank" }
        require(bytes >= 0) { "bytes must not be negative" }
    }
}
